src/inspect_ai/model/_providers/util/batch.py

```diff
@@ -298,19 +299,49 @@ async def _handle() -> dict[str, ResponseT | Exception]:
             return await self._handle_batch_result(batch, completion_info)
 
         try:
-            results = await _handle()
             log_batch(f"Batch {batch.id} completed")
-            # TODO: We don't have any evidence that this actually happens. I
-            # think we should just get rid of the code.
-            if len(results) != len(batch.requests):
-                log_batch(
-                    f"Batch {batch.id} returned {len(results)} results, expected {len(batch.requests)}",
-                )
+            await self._resolve_inflight_batch(batch, await _handle())
+        except Exception as e:
+            await self._fail_and_cleanup_inflight_batch("obtaining results", batch, e)
+
+    async def _resolve_inflight_batch(
+        self, batch: Batch[ResponseT], results: dict[str, ResponseT | Exception]
+    ) -> None:
```

Comment:
This isn't new code. I simply broke it out so that it can be called by the subclass in that case where the entire batch is rejected.